### PR TITLE
Add command support

### PR DESCRIPTION
**Pull request by elenatorro**
Related issues:
* #15 Add command to list offers
* #19 Add help command

### CODE REVIEW

File: src/jobs/jobs.router.js [hunk 1]
```diff
@@ -2,18 +2,85 @@ const winston = require('winston');
...
 const teamsController = require('../teams/teams.controller');
-const jobService = require('./jobs.service');
+const jobsUtils = require('./jobs.utils');
+const jobsService = require('./jobs.service');
 const Responses = require('../constants/responses');
 
 /**
- * Create a new job job.
- * If the job already exists in the database, just broadcast to the channel
+ * Handle /job post command
```
Comment (IagoLast) [inline src/jobs/jobs.router.js]:
```js
/**
 * Handle /job post command. The following commands are availiable.
 * 
 * /job help: Return a message about how to use the bot
 * /job joblist: Return a list of jobs
 * /job: (empty message), return an error
 * /job <url>: Create a new job in the database pointing to the url
 */
```

File: src/jobs/jobs.router.js [hunk 2]
```diff
@@ -2,18 +2,85 @@ const winston = require('winston');
...
  */
 async function post(req, res) {
   winston.info('jobs-router:post', { payload: req.body });
 
+  const slackMessage = req.body.text;
+  const slackCommand = jobsUtils.getCommand(slackMessage);
+
+  return commandFunctions[slackCommand](req, res);
```
Comment (IagoLast) [inline src/jobs/jobs.router.js]: A switch statement makes the intention of this method clear

```js
switch (jobsUtils.getCommand(slackMessage)) {
    case COMMANDS.CREATE_JOB:
      return _createJob(req, res);
    case COMMANDS.HELP:
      return _help(req, res);
...
    default:
      break;
```

File: src/jobs/jobs.router.js [hunk 3]
```diff
@@ -2,18 +2,85 @@ const winston = require('winston');
...
+  createjob:  _sendCreateJob,
+  wrongcommand: _sendHelp
+};
+
+
+/**
+ * Create a new job job.
+ * If the job already exists in the database, just broadcast to the channel
+ */
+async function _sendCreateJob(req, res) {
```
Comment (IagoLast) [inline src/jobs/jobs.router.js]: Im not sure about the `send` prefix, could just name them: createJob, help, list and error?

File: src/slack/JobListMessage.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  constructor(list) {
+    winston.info('JobListMessage:constructor', list);
+
+    this.content = {
+      text: this._generateJobList(list),
+      mrkdwn: true
+    };
+  }
+
+  _generateJobList(list) {
```
Comment (IagoLast) [inline src/slack/JobListMessage.js]:
```js
_generateJobList(list) {
    return list.map(job => `- ${job.link}`).join('\n');
}
```

File: src/jobs/jobs.router.js [hunk 5]
```diff
@@ -2,18 +2,85 @@ const winston = require('winston');
...
+    res.json(list);
+  } catch (error) {
+    winston.error('jobs-router:list', { payload: req.body, error });
+    return res
+      .status(Responses.Jobs.Post.ErrorList.CODE)
+      .send(Responses.Jobs.Post.ErrorList.MESSAGE);
+  }
+}
+
+const commandFunctions = {
```
Comment (IagoLast) [inline src/jobs/jobs.router.js]: with the switch statement this is not needed anymore, but if we are going to use this object I think it should be declared at the beginning of the file following [the stepdown rule](https://dzone.com/articles/the-stepdown-rule):

File: src/constants/commands.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+module.exports = Object.freeze({
+  HELP: 'help',
```
Comment (IagoLast) [inline src/constants/commands.js]: I think uppercase strings equal to the keys would give more aspect of constants

File: src/constants/regex.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+module.exports = Object.freeze({
+  HELP: /^help.*$/,
+  EMPTY: /^$/,
+  JOB_LIST: /^joblist.*$/,
```
Comment (IagoLast) [inline src/constants/regex.js]: I think `/job list` would be better than  `/job joblist`

File: src/jobs/jobs.router.js (hunk 2 above)
Comment (elenatorro) [reply src/jobs/jobs.router.js]: It may be clearer, but this one is faster.

File: src/slack/JobListMessage.js (hunk 4 above)
Comment (elenatorro) [reply src/slack/JobListMessage.js]: Nice ✨

**Pull request merged**